### ISSUE TITLE
Wrong raised reportGeneralTypeIssues using numpy.block(...)

### ISSUE THREAD

**Issue by q-wertz** [microsoft/pyright#4268]
**Describe the bug**
Using `numpy.block(…)` pyright reports an error. (mypy does not complain)

**To Reproduce**
1. Create file `test.py`
    ```python
    import numpy as np
    
    a = np.zeros((3, 1))
    b = np.ones((3, 3))
    c = np.array([[1]])
    
    bm = np.block([[a,   b,
                    b.T, c]])
    ```
2. Run `pyright test.py`
3. Error:
    ```
    No configuration file found.
    No pyproject.toml file found.
    stubPath /home/user/Downloads/typings is not a valid directory.
    Assuming Python platform Linux
    Searching for source files
    Found 1 source file
    pyright 1.1.279
    /home/user/Downloads/test.py
    /home/user/Downloads/test.py:8:16 - error: Argument of type "list[list[NDArray[float64] | NDArray[Any]]]" cannot be assigned to parameter "arrays" of type "ArrayLike" in function "block"
      "list[NDArray[float64] | NDArray[Any]]" is incompatible with protocol "_SupportsArray[dtype[Unknown]]"
        "__array__" is not present (reportGeneralTypeIssues)
    1 error, 0 warnings, 0 informations 
    Completed in 1.782sec
    ```
4. Run `mypy test.py`
5. No error:
    ```
    Success: no issues found in 1 source file
    ```

**Expected behavior**
No error reported by pyright.

**Screenshots or Code**
See above.

**VS Code extension or command-line**
Both. 
Command line pyright version: `1.1.279`
VSCode extension pylance version: `v2022.11.40`

**Additional context**

**Comment by erictraut** [comment]
The latest published version of pyright is 1.1.282. This issue was addressed in 1.1.281. If you update to the latest version, you should no longer see this error.

**Comment by q-wertz** [comment]
Ah sorry. You are right the version on the arch repo seems to be outdated.
